### DISCUSSION TITLE
Is there any way to get mapping of param and its ID?

### DISCUSSION THREAD

**Discussion by skhomuti** [pytest-dev/pytest#8237] [Q&A]
I want to get param name and it's current id, for example, in `pytest_runtest_setup` fixture from `item.callspec` attribute.
```python
import pytest


@pytest.fixture(params=[1, 2], ids=['fixture1', 'fixture2'])
def some_fixture(request):
    pass


@pytest.mark.parametrize('param1, param2', [(1, 2), (3, 4)], ids=['first', 'second'])
def test_params(param1, param2, some_fixture):
    pass
```
For these tests i expected list of such dictionaries:
```
{
    'param1': 'first',
    'param2': 'first',
    'some_fixture': 'fixture1'
}
```
the difficulty is how to correctly determine ID of the parameter, in case of multiple parameters and one ID for them

**Comment by RonnyPfannschmidt** [comment]
unfortunately its not exposed in isolation

**Reply by skhomuti** [reply]
I'll try to explain why I need it.
In custom test reports with support for test parameters, such as Allure-framework, you need to display the value of the parameter or fixture with which the test is running. 
Displaying the value directly may not be entirely informative in the case of objects or large data structures in the parameters, but there is an id in Pytest - a convenient and readable tag of each parameter (or set) for the user.

At the moment Allure takes parameters from the CallSpec2 class. 
What do you think about making changes to this class? For example, add a new attribute containing the relationship between ID and argument names. Now ID is simply added to the list.

**Reply by RonnyPfannschmidt** [reply]
right now i don't have the capacity to plan about such a refactoring
we do want to eventually turn test ids into more complex objects at that point the parameter id would turn into a sequence of keys from parameterize calls, 

a good place for the relation of id parts and parameter names is not clear yet

**Reply by ionapjolive** [reply]
It would be helpful to have the id available in isolation within the test in some testing that I am doing. It is used to name directories for artifacts and as an argument in functions called during the test. Of course, I can create it manually, but it would seem cleaner if it was just available as part of the request fixture (it is sort of, if you parse it from out of the name).
